salt/haproxy/init.sls

```diff
@@ -0,0 +1,48 @@
+---
+/etc/haproxy/haproxy.cfg:
+  file.managed:
+    - source: salt://haproxy/haproxy.cfg.jinja
+    - template: jinja
+    - user: root
+    - group: root
+    - mode: 644
+    - makedirs: True
+    - dir_mode: 755
+
+haproxy:
+  file.managed:
+    - name: /etc/kubernetes/manifests/haproxy.yaml
+    - source: salt://haproxy/haproxy.manifest
+    - template: jinja
+    - user: root
+    - group: root
+    - mode: 644
+    - makedirs: True
+    - dir_mode: 755
+{% if "kube-master" in salt['grains.get']('roles', []) %}
+  iptables.append:
+{% else %}
+  iptables.delete:
```

Comment:
Ahhh. The initial implementation I did of this on the ha branch listened on 6443 on all nodes, and forwarded to the open API servers. The local kubelets / kube-proxy all talked to localhost:6443.

In the case of any software load balancer deployed inside a cluster, and with how kube-proxy works, you are going to be listening on nodes that do not host the service and performing a reverse proxy to the appropriate place, so....